pkg/api/admin/openshiftcluster_validatestatic.go

```diff
@@ -26,7 +26,7 @@ func (sv openShiftClusterStaticValidator) validateDelta(oc, current *OpenShiftCl
 	err := immutable.Validate("", oc, current)
 	if err != nil {
 		err := err.(*immutable.ValidationError)
-		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodePropertyChangeNotAllowed, err.Target, err.Message)
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodePropertyChangeNotAllowed, err.Target, err.Message) //nolint:govet
```

Comment:
Could you please explain what the `govet` error is here and why it's not (easily) fixable?

Reply:
govet error: `::printf: non-constant format string in call to github.com/Azure/ARO-RP/pkg/api.NewCloudError (govet)` the `NewCloudError()` constructor uses `fmt.Sprintf()` and the `err.Message` field passed is of non-constant format string. I'm open to change them to something like `api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodePropertyChangeNotAllowed, err.Target, "Error Message: %s", err.Message)` if that's better.

Reply:
Hm. The constructor has the signature:

```go
func NewCloudError(statusCode int, code, target, message string, a ...interface{}) *CloudError {}
```

It then goes on to do `fmt.Sprintf(message, a...)` - judging by the amount of `//nolint:govet` you had to add, mis-using this constructor seems really common. I agree that using it correctly by passing some static format string would be workable, but you may also investigate how often it's used "correctly" (where someone _actually_ passes in a format string and args). If it's not common, you might save time by changing the signature to 

```go
func NewCloudError(statusCode int, code, target, message string) *CloudError {}
```

Let the caller do the formatting as necessary and simply inline the message in the constructor.

Reply:
There're are around 598 instances of these (correct usage).